Return the file descriptor in A from trap_dos_openfile

jimnicholls opened the pull request:
`trap_dos_openfile` opens a file descriptor for the file and it's the programmer's responsibility to close it. In order to close it, they need to know what it is.

The A register was chosen to align with `trap_dos_opendir` and `trap_dos_findfirst`.

gardners commented on src/hyppo/dos.asm:
```diff
@@ -835,6 +835,8 @@ trap_dos_openfile:
         +Checkpoint "trap_dos_openfile <success>"
 
+        lda dos_current_file_descriptor
+        sta hypervisor_a
```
This pattern is showing up in a few places now, we could save bytes by having a return_from_trap_current_file_descriptor_and_success or similar.

gardners reviewed (changes requested):
Content is fine, just thinking about saving space throughout the hypervisor code by re-using where possible.

jimnicholls commented:
I introduced `return_from_trap_with_success_and_file_descriptor_in_a` just above `return_from_trap_with_success` in main.asm. It does the lda and sta and then falls through to `return_from_trap_with_success`. And I updated `trap_dos_opendir`, `trap_dos_openfile` and `trap_dos_findfirst` to jump to it instead of `return_from_trap_with_success`.

gurcei reviewed (approved):
I've tried this on the hardware with a HICKUP.M65, behaves fine in freezer diskchooser (which makes use of these traps), so happy to merge this one in, thx again! :)
